### PR TITLE
Refactor Time Series Experiment setup method

### LINKED ISSUE #2033: Refactor time series setup
Currently, the setup function is all over the place. Refactor it and organize it in appropriate sections so it is easier to organize and code collaboratively with other developers.

### PR DESCRIPTION
**Pull request by ngupta23**
## Related Issuse or bug

#2033 

Closes #2033

#### Describe the changes you've made

Refactored in a more organized way

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Unit tests to be rerun on GitHub

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: pycaret/internal/pycaret_experiment/time_series_experiment.py [hunk 1]
```diff
@@ -851,6 +851,241 @@ def check_fh(self, fh: Union[List[int], int, np.array]) -> np.array:
...
+
+        ## Make a local copy (to perfrom inplace operation on the original dataset)
+        data_ = data.copy()
+
+        if isinstance(data_, pd.DataFrame):
+            if data_.shape[1] != 1:
+                raise ValueError(
+                    f"Currently only Univariate forecasting without exogenous variables is supported. ",
+                    f"Hence if data is a pandas DataFrame, it must have only 1 column. Got {data_.shape[1]} columns!",
+                )
```
Comment (Yard1) [inline pycaret/internal/pycaret_experiment/time_series_experiment.py:889]:
```suggestion
        if isinstance(data_, pd.DataFrame):
            if data_.shape[1] != 1:
                raise ValueError(
                    f"Currently only Univariate forecasting without exogenous variables is supported. ",
                    f"Hence if data is a pandas DataFrame, it must have only 1 column. Got {data_.shape[1]} columns!",
                )
            ## Make a local copy (to perfrom inplace operation on the original dataset)
            data_ = data.copy()
```

Comment (Yard1) [reply pycaret/internal/pycaret_experiment/time_series_experiment.py:889]: no need for copy if a df is created from series

**Review by Yard1 — APPROVED** [review]
This looks really good, thanks! Just one nit

File: pycaret/internal/pycaret_experiment/time_series_experiment.py (hunk 1 above)
Comment (ngupta23) [reply pycaret/internal/pycaret_experiment/time_series_experiment.py:889]: Well, we are changing the series name so want to avoid the inplace operation.

**Pull request merged**